CDS/scripts/check_date.pl

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/perl
+
+my $version='$Rev: 1 $ $LastChangedDate: 2020-05-15 16:06:33 +0100 (Fri, 15 May 2020) $';
+
+#Module to check dates are not before 2008 and set them to the current date/time if they are.
+#Expects:
+#<key>blah 	   - the name of the key to check for the date
+#END DOC
+
+use CDS::Datastore;
+use File::Slurp qw/read_file/;
+use DateTime::Format::Strptime;
+use DateTime;
+use Date::Manip qw(ParseDate);
+
+#START MAIN
+my $store=CDS::Datastore->new('check_date');
+
+$store->{'debug'}=$ENV{'debug'};
+
+foreach(qw/key/){
+	if(not defined $ENV{$_}){
+		print "-ERROR - you need to specify <$_> to use this module.\n";
+		exit 1;
+	}
+}
+
+my $key=$ENV{'key'};
+if($key eq ''){
+	print "-ERROR - you need to specify a value in <key> to set a metadata key.\n";
+	exit 1;
+}
+
+my @keyparts=split /:/,$ENV{'key'};
+#if no section is specified, default to "meta"
+if(scalar @keyparts==1){
+	$keyparts[1]=$ENV{'key'};
+	$keyparts[0]="meta";
+	$key="meta:$key";
+}
+
+if($keyparts[0] eq "track"){
+	splice @keyparts,1,0,'type';
+}
+
+my $existing_value=$store->get(@keyparts,undef);
+
+my $value=$existing_value;
+my $finalstring=$existing_value;
+my $test_date = ParseDate($value);
```

Comment:
why are we parsing the date twice? surely this makes line 56 un-necessary now

Reply:
The new line parses the date to see if we should log an error and exit.

The old line parses the date for a later comparaison. I think the two libaries I have used for these are likely not compatable so I have kept it seperate.

I did look into trying to work out if the old date parse code worked. Sadly execption catching in Perl appears to be rudimentary compared to other languages so I thought it best to write some other code to handle the new function you want because it was more likely to work.

Reply:
I still don't understand why we can't do the comparison entirely using Date::Manip (i.e. change line 57 to parse the reference date in the same way) and drop out the old date parse entirely?

Reply:
Because one bit of code checks if the string is a date and the other bit of code checks if the date is before 1/1/2008.

Reply:
but the first one has parsed it into some kind of a date that we can do the math on

Reply:
Possibly, but I think it a good idea to keep the code for those two functions separate so one does not break the other.